tensorflow-framework/src/main/java/org/tensorflow/framework/data/impl/MapIterator.java

```diff
@@ -0,0 +1,32 @@
+package org.tensorflow.framework.data.impl;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.data.DatasetIterator;
+import org.tensorflow.framework.data.DatasetOptional;
+import org.tensorflow.op.Ops;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+public class MapIterator extends DatasetIterator {
+  private final BiFunction<Ops, List<Operand<?>>, List<Operand<?>>> mapper;
+
+  public MapIterator(
+      DatasetIterator source,
+      BiFunction<Ops, List<Operand<?>>, List<Operand<?>>> mapper) {
+    super(source);
+    this.mapper = mapper;
+  }
+
+  @Override
+  public List<Operand<?>> getNext() {
+    return mapper.apply(tf, super.getNext());
```

Comment:
Where does `tf` come from here?

Reply:
It's a protected field of the superclass `DatasetIterator`